http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/LeonardoService.scala

```diff
@@ -405,43 +452,49 @@ class LeonardoService(protected val dataprocConfig: DataprocConfig,
                   )
             )
           }
-        } yield true
+        } yield UpdateResult(true, false, Noop(None))
 
-      case None => IO.pure(false)
+      case None => IO.pure(UpdateResult(false, false, Noop(None)))
     }
   }
 
-  def maybeChangeMasterMachineType(existingCluster: Cluster, machineConfigOpt: Option[MachineConfig]): IO[Boolean] = {
+  def maybeChangeMasterMachineType(existingCluster: Cluster, clusterRequest: ClusterRequest): IO[UpdateResult] = {
+    val machineConfigOpt: Option[MachineConfig] = clusterRequest.machineConfig
+    val allowStop: Boolean = clusterRequest.allowStop.getOrElse(false)
+
     val updatedMasterMachineTypeOpt = machineConfigOpt.flatMap { machineConfig =>
       getUpdatedValueIfChanged(existingCluster.machineConfig.masterMachineType, machineConfig.masterMachineType)
     }
 
     updatedMasterMachineTypeOpt match {
       // Note: instance must be stopped in order to change machine type
-      // TODO future enchancement: add capability to Leo to manage stop/update/restart transitions itself.
       case Some(updatedMasterMachineType) if existingCluster.status == Stopped =>
         for {
-          _ <- log.info(
-            s"New machine config present. Changing machine type to ${updatedMasterMachineType} for cluster ${existingCluster.projectNameString}..."
-          )
-          // Update the machine type in Google
-          _ <- clusterHelper.setMasterMachineType(existingCluster, MachineType(updatedMasterMachineType))
-          // Update the DB
-          now <- IO(Instant.now)
-          _ <- clusterQuery
-            .updateMasterMachineType(existingCluster.id, MachineType(updatedMasterMachineType), now)
-            .transaction
-        } yield true
-
-      case Some(_) =>
-        IO.raiseError(ClusterMachineTypeCannotBeChangedException(existingCluster))
+          _ <- clusterHelper.updateMasterMachineType(existingCluster, MachineType(updatedMasterMachineType))
+        } yield UpdateResult(true, false, Noop(None))
+
+      case Some(updatedMasterMachineType) =>
+        logger.info("in stop and update case of maybeChangeMasterMachineType")
+        //        Future.failed(ClusterMachineTypeCannotBeChangedException(existingCluster))
```

Comment:
Remove commented line.
Also should some of these `info` statements be `debug`?

Reply:
fixed them up and added more helpful cluster info to them. I think it will be useful for the initial phases to have as much logging as possible in case anything goes wrong